bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/generator/StatementGenerator.xtend

```diff
@@ -480,29 +482,22 @@ class StatementGenerator {
 		'''(&«e.variable.code»)'''
 	}
 
+	@Traced dispatch def IGeneratorNode code(ReturnValueExpression stmt) {
+		return '''
+			«_code(stmt as AssignmentExpression)»
+			«codeReturnStatement(stmt)»
+		'''
+	}
+
 	@Traced dispatch def IGeneratorNode code(ReturnStatement stmt) {
-		val hasValue = stmt.value !== null;
-		var AbstractTypeGenerator _generatedTypeGenerator = null;
-		var AbstractType _resultType = null;
-		if(hasValue) {
-			val value = stmt.value;
-			val expressionType = BaseUtils.getType(value);
-			if (isGeneratedType(stmt.eResource, expressionType)) {
-				_generatedTypeGenerator = registry.getGenerator(stmt.eResource, expressionType).castOrNull(AbstractTypeGenerator);
-				_resultType = expressionType;
-			}
-		}
-		val generatedTypeGenerator = _generatedTypeGenerator;
-		val resultType = _resultType;
-		
 		'''
-			«IF hasValue && generatedTypeGenerator !== null»
-				«generatedTypeGenerator.generateExpression(resultType, stmt, Optional.absent, codeFragmentProvider.create('''(*_result)'''), codeFragmentProvider.create('''_result'''), AssignmentOperator.ASSIGN, stmt.value).noTerminator»;
```

Comment:
that's really nice to see this going away - it's always been a bit of a sore spot.